Sources/AltSwiftUI/Source/UIKitViews/UIKitCollections.swift

```diff
@@ -139,4 +182,275 @@ class SwiftUIStackView: UIStackView, UIKitViewHandler {
+        while arrangedSubviews.count > movingIndex {
+            let uiView = arrangedSubviews[movingIndex]
+            if !uiView.isHidden {
+                return (uiView: uiView, skippedSubViews: movingIndex - index)
+            }
+            movingIndex += 1
+        }
+        return nil
+    }
+    func setStackAlignment(alignment: HorizontalAlignment) {
+        switch alignment {
+        case .leading:
+            self.alignment = .leading
+        case .center:
+            self.alignment = .center
+        case .trailing:
+            self.alignment = .trailing
+        }
+    }
+    func setStackAlignment(alignment: VerticalAlignment) {
+        switch alignment {
+        case .top:
+            self.alignment = .top
+        case .center:
+            self.alignment = .center
+        case .bottom:
+            self.alignment = .bottom
+        }
+    }
+}
+
+class SwiftUILazyStackView: SwiftUIStackView {
+    var isPendingFirstLayout = true
+    var viewsLengthSum: CGFloat = 0
+    var lastContext: Context?
+    weak var lazyStackScrollView: UIScrollView?
+    /// Content views totally flattened with optional view information
+    var lazyStackFlattenedContentViews: [View] = []
+    var insertLazyContentOnFirstLayout = false
+    var lastInsertedIndex: Int {
+        nonHiddenSubviewsCount - 1
+    }
+    var lazyStackLastEdge: CGFloat {
+        if axis == .horizontal {
+            let visibleLength = lazyStackScrollView?.bounds.width ?? 0
+            let offset = lazyStackScrollView?.contentOffset.x ?? 0
+            return visibleLength + offset - lazyStackOffsetInContainer.minX
+        } else {
+            let visibleLength = lazyStackScrollView?.bounds.height ?? 0
+            let offset = lazyStackScrollView?.contentOffset.y ?? 0
+            return visibleLength + offset - lazyStackOffsetInContainer.minY
+        }
+    }
+    var lazyStackOffsetInContainer: CGRect {
+        guard let containerView = lazyStackScrollView?.subviews.first else {
+            return .zero
+        }
+        return superview?.convert(frame, to: containerView) ?? .zero
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        if isPendingFirstLayout {
+            isPendingFirstLayout = false
+            if insertLazyContentOnFirstLayout {
+                insertLazyViews()
+            }
+        }
+    }
+    override var intrinsicContentSize: CGSize {
+        if lazyStackScrollView != nil {
+            var superSize = super.intrinsicContentSize
+            if axis == .horizontal {
+                superSize.height = CGFloat.limitForUI
+            } else {
+                superSize.width = CGFloat.limitForUI
+            }
+            return superSize
+        } else {
+            return super.intrinsicContentSize
+        }
+    }
+    
+    /// Inserts views since the last inserted index until the last view of the visible area
+    func insertLazyViews() {
```

Comment:
Might be better to just call this `insertViews`. When i read this first i thought that the function inserted only the views that haven't been rendered yet (out of the scroll boundaries) but was also confused because the doc says "until the last view of the visible area"